src/flow/src/adapter/worker.rs

```diff
@@ -276,7 +254,7 @@ impl<'s> Worker<'s> {
     /// Run the worker, blocking, until shutdown signal is received
     pub fn run(&mut self) {
         loop {
-            let (req_id, req) = if let Some(ret) = self.itc_server.blocking_lock().blocking_recv() {
+            let (req, ret_tx) = if let Some(ret) = self.itc_server.blocking_lock().blocking_recv() {
```

Comment:
If we call `run()` in an async runtime then `blocking_lock()` will panic. Is it an expected behavior?

Reply:
> If we call `run()` in an async runtime then `blocking_lock()` will panic. Is it an expected behavior?

Worker is expected to run in a separate sync thread without an async runtime and blocking on idle so yes it's expected(since worker is not send)